test/functional/ui/searchhl_spec.lua

```diff
@@ -148,11 +149,7 @@ describe('search highlighting', function()
   end)
 
   it('is preserved during :terminal activity', function()
-    if iswin() then
-      feed([[:terminal for /L \%I in (1,1,5000) do @(echo xxx & echo xxx & echo xxx)<cr>]])
-    else
-      feed([[:terminal for i in $(seq 1 5000); do printf 'xxx\nxxx\nxxx\n'; done<cr>]])
-    end
+    feed([[:terminal "]]..nvim_dir..[[/shell-test" REP 5000 foo<cr>]])
```

Comment:
Since the screen state is collected by `screen:expect_unchanged()`, we don't need a deterministic initial state (which would then be hardcoded into the test).

This allows us to check "did anything change in the last N ms?" rather than "did anything change compared to a hardcoded screen-state".